src/puppetlabs/comidi.clj

```diff
@@ -63,9 +70,9 @@
       (str/replace #"-$" "")))
 
 (defn special-chars->underscores
-  "Convert all non-alpha chars except * and - to underscores"
+  "Convert all non-alpha chars except ! * and - to underscores"
   [s]
-  (str/replace s #"[^\w\*\-]" "_"))
+  (str/replace s #"[^\w\!\*\-]" "_"))
```

Comment:
"false" support seems kind of odd but it felt necessary to balance out the "true" changes.